plugins/by-name/overseer/default.nix

```diff
@@ -0,0 +1,132 @@
+    '';
+
+    dap = defaultNullOpts.mkBool true ''
+      Whether to patch nvim-dap to support preLaunchTask and postDebugTask
+    '';
+
+    task_list =
+      defaultNullOpts.mkAttrsOf types.anything
+        {
+          default_detail = 1;
+          max_width = [
+            100
+            0.2
+          ];
+          min_width = [
+            40
+            0.1
+          ];
+          width = null;
+          max_height = [
+            20
+            0.1
+          ];
+          min_height = 8;
+          height = null;
+          separator = "────────────────────────────────────────";
+          direction = "bottom";
+          bindings = {
+            "?" = "ShowHelp";
+            "g?" = "ShowHelp";
+            "<CR>" = "RunAction";
+            "<C-e>" = "Edit";
+            "o" = "Open";
+            "<C-v>" = "OpenVsplit";
+            "<C-s>" = "OpenSplit";
+            "<C-f>" = "OpenFloat";
+            "<C-q>" = "OpenQuickFix";
+            "p" = "TogglePreview";
+            "<C-l>" = "IncreaseDetail";
+            "<C-h>" = "DecreaseDetail";
+            "L" = "IncreaseAllDetail";
+            "H" = "DecreaseAllDetail";
+            "[" = "DecreaseWidth";
+            "]" = "IncreaseWidth";
+            "{" = "PrevTask";
+            "}" = "NextTask";
+            "<C-k>" = "ScrollOutputUp";
+            "<C-j>" = "ScrollOutputDown";
+            "q" = "Close";
+          };
+        }
+        ''The task list displays all tasks that have been created. It shows the task status, name, and a summary of the task output.'';
+
+    task_editor = {
+      bindings =
+        defaultNullOpts.mkAttrsOf (with types; either str (enum [ false ]))
+          {
+            i = {
+              "<CR>" = "NextOrSubmit";
+              "<C-s>" = "Submit";
+              "<Tab>" = "Next";
+              "<S-Tab>" = "Prev";
+              "<C-c>" = "Cancel";
+            };
+            n = {
+              "<CR>" = "NextOrSubmit";
+              "<C-s>" = "Submit";
+              "<Tab>" = "Next";
+              "<S-Tab>" = "Prev";
+              "q" = "Cancel";
+              "?" = "ShowHelp";
+            };
+          }
+          ''
+            Set keymap to false to remove default behavior
+            You can add custom keymaps here as well (anything vim.keymap.set accepts)
+          '';
+    };
+
+    task_launcher = {
+      bindings =
+        defaultNullOpts.mkAttrsOf (with types; either str (enum [ false ]))
+          {
+            i = {
+              "<C-s>" = "Submit";
+              "<C-c>" = "Cancel";
+            };
+            n = {
+              "<CR>" = "Submit";
+              "<C-s>" = "Submit";
+              "q" = "Cancel";
+              "?" = "ShowHelp";
+            };
+          }
+          ''
+            Set keymap to false to remove default behavior
+            You can add custom keymaps here as well (anything vim.keymap.set accepts)
```

Comment:
Ensure to end all sentences with a period.
Like so, we can provide a consistent documentation style.